clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -12383,6 +12383,13 @@ def warn_unsafe_buffer_operation : Warning<
   "%select{unsafe pointer operation|unsafe pointer arithmetic|"
   "unsafe buffer access|function introduces unsafe buffer manipulation|unsafe invocation of span::data}0">,
   InGroup<UnsafeBufferUsage>, DefaultIgnore;
+def warn_unsafe_buffer_libc_call : Warning<
+  "function %0 introduces unsafe buffer access">,
+  InGroup<UnsafeBufferUsage>, DefaultIgnore;
+def note_unsafe_buffer_printf_call : Note<
+  "%select{| change to 'snprintf' for explicit bounds checking | buffer pointer and size may not match"
+          "| use 'std::string::c_str' or string literal as string pointer to guarantee null-termination"
```

Comment:
Also if `-Wunsafe-buffer-usage` is enabled in C (for any obscure reason), would we still recommend `c_str` here? Maybe this recommendation should be guarded by the "emit suggestions" flag?

Reply:
If we want to keep this patch minimal then I suggest splitting the note off as a follow-up improvement.

Reply:
Communicating what parameters are used in an unsafe way wouldn't be strictly necessary for the initial patch if the warning as is pushed the user to do the right thing.
Have we tried getting data from a real project?

I can also imagine that for some functions it won't be as simple as saying "pointer parameter 1, 3 and 5" as it could be interplay between pointers, integer parameters and buffer content.

Reply:
> Have we tried getting data from a real project?

Yes! Basically these special handlings are all corresponding to actual cases in our downstream projects.

>I think for this warning in particular, it's valuable to point the user to the specific %s argument. To, at least, make sure that they know we don't mean the snprintf's target string or something.

I will let the notes point to specific argument source locations.  For better note messages, we could do it in a follow-up patch?

Reply:
> I can also imagine that for some functions it won't be as simple as saying "pointer parameter 1, 3 and 5" as it could be interplay between pointers, integer parameters and buffer content.

Not sure if I understand your concern.  Could you give an example?